catalogue_pipeline/reindexer_v2/reindex_shard_generator/src/reindex_shard_generator.py

```diff
@@ -0,0 +1,64 @@
+# -*- encoding: utf-8
+
+import os
+
+import boto3
+
+from shard_manager import create_reindex_shard
+
+from wellcome_aws_utils.sns_utils import extract_sns_messages_from_lambda_event
+
+
+SOURCE_SIZES = {
+    'sierra': 5e7,
+    'miro': 2.5e5,
+}
+
+
+def main(event, _ctxt=None, dynamodb_client=None):
+    print(f'event={event!r}')
+
+    dynamodb_client = dynamodb_client or boto3.client('dynamodb')
+    table_name = os.environ['TABLE_NAME']
+
+    for sns_event in extract_sns_messages_from_lambda_event(event):
+        row = sns_event.message
+
+        # If we've already looked at this row before, there's nothing
+        # for us to do -- we can skip to the next row.
+        if row.get('reindexShard') != 'default':
```

Comment:
It still might be worth checking that the shard we generate for this ID matches the one given?

Reply:
Good point; I’ll change it.